### PR TITLE
CMake swtich for audio support

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: Michał Szołucha <mszolucha@nvidia.com>

#### Why we need this PR?
*Pick one*
- Adds a compilation swtich for audio, so that you can build DALI for a single use-case, not for all of them at the same time

### CODE REVIEW

**Comment by szalpal** [comment]
!build

File: conda/recipe/meta.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ build:
    - BUILD_LIBTIFF
    - BUILD_NVOF
    - BUILD_NVDEC
+   - BUILD_LIBSND
```
Comment (JanuszL) [inline conda/recipe/meta.yaml]: You need to add https://anaconda.org/conda-forge/libsndfile as a build and run dependency as well here.

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -42,9 +42,6 @@ if (BUILD_NVJPEG)
   if (${NVJPEG_DECOUPLED_API})
     add_definitions(-DNVJPEG_DECOUPLED_API)
   endif()
-else()
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: Why you are removing this statuses?

File: conda/recipe/meta.yaml (hunk 1 above)
Comment (jantonguirao) [reply conda/recipe/meta.yaml]: Will be addressed here https://github.com/NVIDIA/DALI/pull/1483

File: cmake/Dependencies.cmake (hunk 2 above)
Comment (szalpal) [reply cmake/Dependencies.cmake]: With @klecki we figured out, it's gonna be better, to keep them in one place, `propagate_option` to be precise. I added it there in this PR also

Comment (klecki) [reply cmake/Dependencies.cmake]: Yup, we now log the options when we propagate them.

Comment (klecki) [reply cmake/Dependencies.cmake]: So all of them are listed.

File: docker/Dockerfile.build.aarch64-linux [hunk 3]
```diff
@@ -175,6 +175,7 @@ CMD cmake \
   -DBUILD_NVJPEG=OFF \
   -DBUILD_NVOF=OFF \
   -DBUILD_NVDEC=OFF \
+  -DBUILD_LIBSND=OFF \
```
Comment (jantonguirao) [inline docker/Dockerfile.build.aarch64-linux]: Any reason for it to be off here and on in QNX?

Comment (szalpal) [reply docker/Dockerfile.build.aarch64-linux]: Turned on

File: CMakeLists.txt [hunk 4]
```diff
@@ -34,6 +34,7 @@ option(BUILD_LIBTIFF "Build with libtiff support" ON)
 option(BUILD_NVJPEG "Build with nvJPEG support" ON)
 option(BUILD_NVOF "Build with NVIDIA OPTICAL FLOW SDK support" ON)
 option(BUILD_NVDEC "Build with NVIDIA NVDEC support" ON)
+option(BUILD_LIBSND "Build with audio support" ON)
```
Comment (jantonguirao) [inline CMakeLists.txt]:
```suggestion
option(BUILD_LIBSND "Build with suport for libsnd libraryt" ON)
```

**Comment by szalpal** [comment]
!build

**Pull request merged**